legend-sdlc-server/src/main/java/org/finos/legend/sdlc/server/gitlab/auth/GitLabUserContext.java

```diff
@@ -110,7 +112,14 @@ else if (redirectAllowed)
                             throw new LegendSDLCServerException("{\"message\":\"Authorization required\",\"auth_uri\":\"/auth/authorize\"}", Status.FORBIDDEN);
                         }
                     }
-                    api = new GitLabApi(ApiVersion.V4, modeInfo.getServerInfo().getGitLabURLString(), TokenType.OAUTH2_ACCESS, accessToken);
+                    else
+                    {
+                        if (gitLabSession instanceof GitLabPersonalAccessTokenSession && !token.getToken().equals(this.httpRequest.getHeader("PRIVATE-TOKEN")))
+                        {
+                            throw new LegendSDLCServerException("{\"message\":\"Private token should not change in one session\"}", Status.FORBIDDEN);
+                        }
+                    }
```

Comment:
I'm not sure this check is necessary, at least not here. If we're concerned about the token changing during a session, we should be checking that during the authentication stage.